api/channel.go

```diff
@@ -965,7 +866,27 @@ func getChannel(c *Context, w http.ResponseWriter, r *http.Request) {
 			w.Write([]byte(data.ToJson()))
 		}
 	}
+}
 
+func SetActiveChannel(userId string, channelId string) *model.AppError {
+	status, err := GetStatus(userId)
+	if err != nil {
+		status = &model.Status{userId, model.STATUS_ONLINE, false, model.GetMillis(), channelId}
+	} else {
+		status.ActiveChannel = channelId
+		if !status.Manual {
+			status.Status = model.STATUS_ONLINE
+		}
+		status.LastActivityAt = model.GetMillis()
+	}
+
+	AddStatusCache(status)
+
+	if result := <-Srv.Store.Status().SaveOrUpdate(status); result.Err != nil {
```

Comment:
Do we really have to send this to the DB? 

How about only sending to the DB for manual statuses? You can update the cache when the server starts and every time anyone changes their manual status. Automatic status is probably the common case.